src/filesystem/drive/inner.rs

```diff
@@ -95,15 +120,36 @@ impl InnerDrive {
 
             // Update Size:
             let new_children_size = node.ordered_child_pids().iter().fold(0, |acc, child_pid| {
+                // This should maybe actually float up an error as opposed to defaulting to 0 if a child can not be found using `try_fold`
+                // It implies a child on the current node can't be found in the filesystem.
+                // This should be done carefully though, maybe finishing the rest of its work before returning an error? Or maybe not?
```

Comment:
Highlighting this as a discussion point

Reply:
Yeah I think any time we have a reference to a permanent ID that isn't present we need to fail hard and fast. We want to preserve only known good instances of the filesystem especially in the early days. If this ever occurs we've lost data and we need to rollback any operations done since the last sync, log and address the issue.

There can be zero silent corruption errors the code base.

Reply:
Alright I'm using `try_fold` and it will propagate any errors now